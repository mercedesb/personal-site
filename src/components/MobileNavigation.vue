<template>
  <div class='MobileNav' v-on:click="expanded = !expanded">
    <div class='MobileNav-hamburger'>
      <img src='../assets/menu.svg' alt='hamburger menu'/>
    </div>
    <transition name='grow'>
      <ul class='MobileNav-navLinks' v-if="navLinks.length && expanded">
        <li class="MobileNav-navLink">
          <smart-link to="/">
            Home
          </smart-link>
        </li>
        <li class="MobileNav-navLink" v-for="navLink in navLinks" :key="navLink.title">
          <smart-link
            :to="navLink.parsedLink"
            :isExternal="!!navLink.externalLink"
            >
            {{navLink.title}}
          </smart-link>
        </li>
      </ul>
    </transition>
  </div>
</template>

<script>
export default {
  props: {
    navLinks: {
      type: Array,
      required: true
    }
  },
  data () {
    return {
      expanded: false
    }
  },
  watch: {
    '$route' (to, from) {
      this.expanded = false
    }
  }
}
</script>

<style lang="scss">
@import '../assets/styles/variables.scss';

.MobileNav {
  display: flex;
  flex-direction: column;
  position: relative;
  width: 100%;

  img {
    height: 50px;
    &:hover {
      cursor: pointer;
    }
  }

  @include media($min-desktop) {
    display: none;
  }

  &-hamburger {
    width: 100%;
    display: flex;
    flex-direction: row-reverse;
  }

  &-navLinks {
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-bottom: 0px;
  }

  &-navLink {
    padding: $small-spacing 0;
    a {
      color: $white;
      font-size: $large-font-size;
    }
  }
}
</style>
