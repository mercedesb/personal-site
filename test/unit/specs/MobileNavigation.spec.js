import MobileNavigation from '@/components/MobileNavigation.vue'
import { TestUtility } from './TestUtility'

describe('MobileNavigation', () => {
  let component

  const initialProps = {
    backgroundImages: [
      {
        file: {
          url: 'http://placeholder.pics/svg/300'
        }
      },
      {
        file: {
          url: 'http://placeholder.pics/svg/300'
        }
      },
      {
        file: {
          url: 'http://placeholder.pics/svg/300'
        }
      }
    ],
    navLinks: Object.keys(TestUtility.landingPages).map((key) => {
      return TestUtility.landingPages[key]
    })
  }

  const shallow = propsData => TestUtility.shallow(MobileNavigation, {
    propsData: {
      ...initialProps,
      ...propsData
    }
  })

  describe('Snapshots', () => {
    it('matches snapshot', () => {
      component = shallow()
      expect(component.element).toMatchSnapshot()
    })
  })
})
