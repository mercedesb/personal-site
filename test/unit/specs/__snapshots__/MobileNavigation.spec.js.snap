// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavigation Snapshots matches snapshot 1`] = `
<div
  class="MobileNav"
>
  <div
    class="MobileNav-hamburger"
  >
    <img
      alt="hamburger menu"
      src="../assets/menu.svg"
    />
  </div>
   
  <!---->
</div>
`;
